core/src/main/scala/org/apache/spark/deploy/rest/mesos/MesosRestServer.scala

```diff
@@ -94,7 +94,12 @@ private[mesos] class MesosSubmitRequestServlet(
     val driverMemory = sparkProperties.get("spark.driver.memory")
     val driverCores = sparkProperties.get("spark.driver.cores")
     val appArgs = request.appArgs
-    val environmentVariables = request.environmentVariables
+    // We don't want to pass down SPARK_HOME when launching Spark apps
+    // with Mesos cluster mode since it's populated by default on the client and it will
+    // cause spark-submit script to look for files in SPARK_HOME instead.
+    // We only need the ability to specify where to find spark-submit script
+    // which user can user spark.executor.home or spark.home configurations.
+    val environmentVariables = request.environmentVariables.filter(!_.equals("SPARK_HOME"))
```

Comment:
Hmm interesting I wonder if I ran it differently than what my code have, since I was able to see it not passed through.
Thanks for retesting this, I think having the automated tests is going to be crucial to prevent mistakes like this that I'm making :(